azure/functions/decorators/function_app.py

```diff
@@ -1373,3 +1375,94 @@ def decorator():
             return decorator()
 
         return wrap
+
+    def event_grid_trigger(self,
+                           arg_name: str,
+                           data_type: Optional[
+                               Union[DataType, str]] = None,
+                           **kwargs) -> Callable:
+        """
+        The event_grid_trigger decorator adds
+        :class:`EventGridTrigger`
+        to the :class:`FunctionBuilder` object
+        for building :class:`Function` object used in worker function
+        indexing model. This is equivalent to defining event grid trigger
+        in the function.json which enables function to be triggered to
+        respond to an event sent to an event grid topic.
+        All optional fields will be given default value by function host when
+        they are parsed by function host.
+
+        Ref:
+        https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-event-grid-trigger
+
+        :param arg_name: the variable name used in function code for the
+         parameter that receives the event data.
+        :param data_type: Defines how Functions runtime should treat the
+        parameter value.
+        :return: Decorator function.
+        """
+
+        @self._configure_function_builder
+        def wrap(fb):
+            def decorator():
+                fb.add_trigger(
+                    trigger=EventGridTrigger(
+                        name=arg_name,
+                        data_type=parse_singular_param_to_enum(data_type,
+                                                               DataType),
+                        **kwargs))
+                return fb
+
+            return decorator()
+
+        return wrap
+
+    def write_event_grid(self,
```

Comment:
Did we finalize of this name? write_event_grid_message would be more appropriate here IMO. Thoughts? @YunchuWang @vrdmr @shreyabatra4

Reply:
as write_event_hub_message is what we are doing for event hub, write_event_grid_message may be more aligned

Reply:
updated as per recommendation.